src/ipc/ipc-helper.c

```diff
@@ -157,21 +156,29 @@ int comp_verify_params(struct comp_dev *dev, uint32_t flag,
 
 		while (clist != buffer_list) {
 			curr = clist;
```

Comment:
don't component buffer lists have to be protected? I still don't understand our design of component, buffer and other element sharing well enough, but it seems like it can be or become a problem? I see buffers are added to the lists in `pipeline_connect()` with just local interrupts disabled, but probably even with dynamic pipelines it cannot run simultaneously on different cores? Say, if you have a mixer, where one core is running a pipeline. Then an IPC is arriving to start a pipeline on another core with the same mixer, so that core tries to add a buffer to its list?